File: Module3/examples/inclass/scripts/gamelogic.js
"use strict";
var Game = function (seed) {
    var score = 1000,
        randomSayi = random(seed);

    this.guess = function (userGuess) {
        var guess =userGuess;
        var diff = Math.abs(guess - randomSayi);
        var feedback = {};

        switch (diff) {
            case 0:
                feedback.status = true;      
                break;
            case 1:
                score -= 20;
                feedback.status = false;
                break;
            case 2:
                score -= 50;
                feedback.status = false;
                break;
            default:
                score -= 100;
                feedback.status = false;
                break;
        }
        if (score <= 0) {
            feedback.status = true;
        }

        feedback.score = score;

        return feedback;
    }

    this.restart = function () {
        score = 1000;
        randomSayi = random(seed);
    };

    function random(seed) {
        return Math.ceil(Math.random() * seed);
    };

    return this;
};

File: Module5/examples/inclass/Module5/BusinessLogic/GameLogic.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;


namespace Module5.BusinessLogic
{
    public class GuessResult
    {
        public bool Status { get; set; }
        public int Score { get; set; }
    }
    public class GameLogic
    {
        public int Seed { private get; set; }
        private int RandomNumber {  get;  set; }
        private int Score { get; set; }
        public GameLogic(int seed)
        {
            Seed = seed;
            Restart();
        }

        public GuessResult Guess(int guess)
        {
            var result = new GuessResult();
            var diff = Math.Abs(RandomNumber - guess);

            switch (diff)
            {
                case 0:
                    result.Status = true;
                    break;
                case 1:
                    result.Status = false;
                    Score -= 20;
                    break;
                case 2:
                    result.Status = false;
                    Score -= 50;
                    break;
                default:
                    result.Status = false;
                    Score -= 100;
                    break;
            }

            if(Score<=0)
                result.Status = true;

            result.Score = Score;

            return result;
        }

        public void Restart()
        {
            Score = 1000;
            RandomNumber = new Random().Next(1, Seed);
        }
    }
}